### PR TITLE
change ci docker image to rocm6.1

### PR DESCRIPTION
**Pull request by kailums**
### Description

There is a bug for kernel running on rocm6.0, so change ci docker image to rocm6.1

For the torch installed in the docker image, change to rocm repo when it is not 6.0 version.

### CODE REVIEW

File: tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile [hunk 1]
```diff
@@ -77,7 +77,7 @@ RUN ln -sf /usr/lib/x86_64-linux-gnu/libstdc++.so.6 ${CONDA_ENVIRONMENT_PATH}/bi
 RUN export MAJOR=$(cut -d '.' -f 1 <<< "$ROCM_VERSION") && \
     export MINOR=$(cut -d '.' -f 2 <<< "$ROCM_VERSION") && \
     export PATCH=$(cut -d '.' -f 3 <<< "$ROCM_VERSION") && \
-    if (( MAJOR >= 6 )); then \
+    if (( MAJOR == 6 && MINOR == 0 )); then \
```
Comment (cloudhan) [inline tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile:80]: gqa test will produce nan you use old torch and triton

File: tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile [hunk 2]
```diff
@@ -77,7 +77,7 @@ RUN ln -sf /usr/lib/x86_64-linux-gnu/libstdc++.so.6 ${CONDA_ENVIRONMENT_PATH}/bi
...
         pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm${MAJOR}.${MINOR} ; \
     else \
         pip install torch==2.0.1 torchvision==0.15.2 -f https://repo.radeon.com/rocm/manylinux/rocm-rel-${MAJOR}.${MINOR}/ ; \
```
Comment (mindest) [inline tools/ci_build/github/pai/rocm-ci-pipeline-env.Dockerfile:83]: Should we also use the latest torch and torchvision under `https://repo.radeon.com/rocm/manylinux/rocm-rel-${MAJOR}.${MINOR}`?
```suggestion
        pip install torch torchvision -f https://repo.radeon.com/rocm/manylinux/rocm-rel-${MAJOR}.${MINOR}/ ; \
```

**Comment by PeixuanZuo** [comment]
please also change the rocm version here:   https://github.com/microsoft/onnxruntime/blob/2c53b4a534a9b64466e435d384c91f0b684ea58a/tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml#L28

**Review by PeixuanZuo — APPROVED** [review]

**Review by mindest — APPROVED** [review]

**Comment by kailums** [comment]
@microsoft/onnxruntime-es please take a look at this PR.

**Review by snnn — APPROVED** [review]

**Pull request merged**